### ISSUE TITLE
TypeError on commit after any change to .ts file in `packages/hardhat`

### ISSUE THREAD

**Issue by MattPereira** [scaffold-eth/scaffold-eth-2#772]
### Is there an existing issue for this?

- [X] I have looked through the [existing issues](https://github.com/scaffold-eth/scaffold-eth-2/issues)

### Which method was used to setup Scaffold-ETH 2 ?

npx create-eth@latest

### Current Behavior

Running into this TypeError when making any change (including just a comment) to any of the typescript files in the packages/hardhat folder

```
[ test ] $ git commit -m 'test deploy script change'
✔️ Preparing lint-staged...
❯ Running tasks for staged files...
  ❯ .lintstagedrc.js — 1 file
    ↓ packages/nextjs/**/*.{ts,tsx} — no files [SKIPPED]
    ❯ packages/hardhat/**/*.{ts,tsx} — 1 file
      ✖️ yarn hardhat:lint-staged --fix deploy/00_deploy_your_contract.ts [FAILED]
↓ Skipped because of errors from tasks. [SKIPPED]
✔️ Reverting to original state because of errors...
✔️ Cleaning up temporary files...

✖️ yarn hardhat:lint-staged --fix deploy/00_deploy_your_contract.ts:

Oops! Something went wrong! :(

ESLint: 8.26.0

TypeError [ERR_INVALID_ARG_TYPE]: Error while loading rule '@typescript-eslint/no-unnecessary-type-constraint': The "path" argument must be of type string. Received undefined
Occurred while linting /home/matthu/Desktop/test/packages/hardhat/deploy/00_deploy_your_contract.ts

```

### Expected Behavior

Since only adding a comment to a deploy script and/or `hardhat.config.ts`, I expected no TypeError that blocks me from committing changes

### Steps To Reproduce

1. `npx create-eth@latest` 
2. choose hardhat framework
3. install packages
4. make any change to any .ts file in the `packages/hardhat` folder ( including just a comment in the deploy script for example )
5. stage and commit the change

### Anything else?

- Started happening about 2 weeks ago
- Happens to me on both my desktop and laptop, both of which run Ubuntu 22.04.3
- Using node v18.18.2

**Comment by technophile-04** [comment]
Thanks @MattPereira for raising the issue ! 

Seem like something is broke with `@typescript-eslint/parser`  it is not being compatible with eslint `8.26.0`

![Screenshot 2024-03-11 at 10 55 08 PM](https://github.com/scaffold-eth/scaffold-eth-2/assets/80153681/2fe78ca9-661e-4fca-840e-26141cb2c66a)
 

We should probably fix the version from `latest` => `~6.7.3`
![Screenshot 2024-03-11 at 10 53 10 PM](https://github.com/scaffold-eth/scaffold-eth-2/assets/80153681/183d6d75-d668-4dc2-9b90-2e53064a542c)

Will create a PR for this 🙌

**Comment by technophile-04** [comment]
Should be fixed in latest version which was just published https://github.com/scaffold-eth/scaffold-eth-2/releases/tag/v0.0.35 

It should now work fine with `npx create-eth@latest` 

Closing this for now, but please free to open new issue you again face the issue 🙌
